### ISSUE TITLE
Bug: Nominatim Search API - featuretype maps to an incorrect query param

### ISSUE THREAD

**Issue by mrsruj** [geopy/geopy#564]
In the [Nominatim geoencoder](https://github.com/geopy/geopy/blob/a3175d7cd6f05e223e60642237bfe297dd88c4a5/geopy/geocoders/nominatim.py#L292C21-L292C32), the `featuretype` parameter is mapped to `"featuretype"` instead of `"featureType"` as required (per the [Nominatim Search API docs](https://nominatim.org/release-docs/latest/api/Search/)).

**This renders the `featuretype` param in the Nominatim geoencoder useless. Here is a test case.**

Search `"new york"` and restrict results to `state` i.e. `featureType=state`. Expected output should be [New York State](https://en.wikipedia.org/wiki/New_York_(state)), geopy returns [New York City](https://en.wikipedia.org/wiki/New_York_City)

- Via Geopy
```python
from geopy.geocoders import Nominatim
geolocator = Nominatim(user_agent="test_featuretype")
location = geolocator.geocode("new york", featuretype="state")
print(location.raw)
# {'place_id': 371663561, 'licence': 'Data © OpenStreetMap contributors, ODbL 1.0. http://osm.org/copyright', 'osm_type': 'relation', 'osm_id': 175905, 'lat': '40.7127281', 'lon': '-74.0060152', 'class': 'boundary', 'type': 'administrative', 'place_rank': 10, 'importance': 0.8175766114518461, 'addresstype': 'city', 'name': 'City of New York', 'display_name': 'City of New York, New York, United States', 'boundingbox': ['40.4765780', '40.9176300', '-74.2588430', '-73.7002330']}
print(location.raw.get("addresstype"))
# city
```
- Via Nominatim Search API Call with **wrong** `featuretype` param (like in geopy)
```bash
curl --location 'https://nominatim.openstreetmap.org/search?q=new%20york&limit=1&format=json&featuretype=state' --header 'User-Agent: test_featuretype'
# [{"place_id":371380173,"licence":"Data © OpenStreetMap contributors, ODbL 1.0. http://osm.org/copyright","osm_type":"relation","osm_id":175905,"lat":"40.7127281","lon":"-74.0060152","class":"boundary","type":"administrative","place_rank":10,"importance":0.8175766114518461,"addresstype":"city","name":"City of New York","display_name":"City of New York, New York, United States","boundingbox":["40.4765780","40.9176300","-74.2588430","-73.7002330"]}]
```

- Via Nominatim Search API Call with **correct** `featureType` param
```bash
curl --location 'https://nominatim.openstreetmap.org/search?q=new%20york&limit=1&format=json&featureType=state' --header 'User-Agent: test_featuretype'
# [{"place_id":306170897,"licence":"Data © OpenStreetMap contributors, ODbL 1.0. http://osm.org/copyright","osm_type":"relation","osm_id":61320,"lat":"43.1561681","lon":"-75.8449946","class":"boundary","type":"administrative","place_rank":8,"importance":0.7655846409089574,"addresstype":"state","name":"New York","display_name":"New York, United States","boundingbox":["40.4765780","45.0158611","-79.7619758","-71.7909720"]}]
```

**Comment by KostyaEsmukov** [comment]
So that's why these tests are failing.. Good catch, thank you! The lowercased key had been working until some time around Nov 2022 -- Aug 2023. Should be fixed now by 71894db

**Comment by ghost** [comment]
Thanks 🙏
